biz.aQute.bnd/src/aQute/bnd/main/ResolveCommand.java

```diff
@@ -268,28 +272,33 @@ public void _resolve(ResolveOptions options) throws Exception {
 			if (!f.isFile()) {
 				error("Missing bndrun file: %s", f);
 			} else {
-
 				Run run = Run.createRun(ws, f);
-
-				try (ProjectResolver pr = new ProjectResolver(run)) {
-					try {
-						Map<Resource,List<Wire>> resolution = pr.resolve();
-						if (pr.isOk()) {
-							System.out.printf("# %-50s ok\n", f.getName());
-							if (options.bundles()) {
-								for (Resource r : resolution.keySet()) {
-									IdentityCapability id = ResourceUtils.getIdentityCapability(r);
-									List<ContentCapability> content = ResourceUtils.getContentCapabilities(r);
-									System.out.printf("  %-50s %40s %s\n", id.osgi_identity(),
-											content.get(0).osgi_content(), content.get(0).url());
+				if (options.write()) {
+					Bndrun bndrun = new Bndrun(run.getWorkspace(), f);
+					String resolved = bndrun.resolve(false, true);
```

Comment:
It's arguably API because somebody could have built a toolchain that post-processes that output, e.g. with awk or sed. I agree it's not a very useful format aside from that -- and also inconsistent with the failure output of the Maven plugins such as resolve-, export- and testing-maven-plugin.